src/amo/components/CollectionList/index.js

```diff
@@ -0,0 +1,174 @@
+        const { authorUsername, id, name, numberOfAddons, slug } = collection;
+        collectionElements.push(
+          <UserCollection
+            authorUsername={authorUsername}
+            id={id}
+            name={name}
+            numberOfAddons={numberOfAddons}
+            slug={slug}
+          />,
+        );
+      });
+    } else {
+      // Create 4 "loading" components.
+      for (let count = 0; count < 4; count++) {
+        collectionElements.push(<UserCollection id={count} loading />);
+      }
+    }
+
+    const footer = collectionElements.length ? null : noCollectionsText;
+
+    return (
+      <CardList
+        className="CollectionList-list"
+        footer={footer}
+        header={i18n.gettext('My collections')}
+      >
+        {collectionElements.length && (
+          <ul className="CollectionList-listing">{collectionElements}</ul>
+        )}
+      </CardList>
+    );
+  }
+
+  render() {
+    const { i18n, isLoggedIn, location } = this.props;
+
+    return (
+      <div className="CollectionList">
+        <div className="CollectionList-wrapper">
+          <Card
+            className="CollectionList-info"
+            header={i18n.gettext('About collections')}
+          >
+            {!isLoggedIn ? (
+              <AuthenticateButton
+                noIcon
+                location={location}
+                logInText={i18n.gettext('Log in to view your collections')}
+              />
+            ) : (
+              <React.Fragment>
+                <p className="CollectionList-info-text">
+                  {i18n.gettext(`Collections make it easy to keep track of favorite
+                    add-ons and share your perfectly customized browser with others.`)}
+                </p>
+                <Button
+                  buttonType="action"
+                  className="CollectionList-create"
+                  puffy
+                  to="/collections/add/"
+                >
+                  {i18n.gettext('Create a collection')}
+                </Button>
+              </React.Fragment>
+            )}
+          </Card>
+          {isLoggedIn ? this.renderCollections() : null}
+        </div>
+      </div>
+    );
+  }
+}
+
+export const mapStateToProps = (state: {|
+  collections: CollectionsState,
+  users: UsersStateType,
+|}) => {
+  const { collections, users } = state;
+
+  const currentUser = getCurrentUser(users);
+  const currentUsername = currentUser && currentUser.username;
+
+  let userCollections;
+
+  if (currentUsername) {
+    userCollections = collections.userCollections[currentUsername];
+  }
+
+  return {
+    currentUsername,
+    isLoggedIn: !!currentUser,
+    loadingUserCollections: userCollections ? userCollections.loading : false,
+    collections: expandCollections(collections, userCollections),
+  };
+};
+
+export const extractId = (ownProps: Props) => {
+  const { currentUsername } = ownProps;
+  return currentUsername || '';
+};
+
+export default compose(
```

Comment:
We should use the pattern described here now: https://github.com/mozilla/addons-frontend#flow

Reply:
Ok, I updated it. It's a bit odd because it doesn't actually have any public `Props`, all props are internal. I still followed the pattern explicitly, which will make it easier to update if we do add a public prop later.